baseplate/context/sqlalchemy.py

```diff
@@ -148,30 +164,17 @@ class SQLAlchemySessionContextFactory(SQLAlchemyEngineContextFactory):
     """
 
     def make_object_for_context(self, name, span):
-        if isinstance(span, ServerSpan):
-            engine = super(SQLAlchemySessionContextFactory, self).make_object_for_context(
-                name, span
-            )
-            session = Session(bind=engine)
-        else:
-            # Reuse session in the existing context
-            #  There should always be one passed down from the
-            #  root ServerSpan
-            session = getattr(span.context, name)
-        span.register(SQLAlchemySessionSpanObserver(session, span))
+        engine = super().make_object_for_context(name, span)
+        session = Session(bind=engine)
+        span.register(SQLAlchemySessionSpanObserver(session))
         return session
 
 
 class SQLAlchemySessionSpanObserver(SpanObserver):
     """Automatically close the session at the end of each request."""
 
-    def __init__(self, session, span):
+    def __init__(self, session):
         self.session = session
-        self.span = span
 
     def on_finish(self, exc_info):
-        # A session is passed down to child local spans
-        #   in a request pipeline so only close the session
-        #  if the parent ServerSpan is closing.
-        if isinstance(self.span, ServerSpan):
```

Comment:
This was kinda broken before and definitely didn't work after these changes, so I just kinda flippantly simplified it. Previously, we kept the same `Session` across to local spans which meant you could safely interact with objects you'd fetched in the outer scope, but the cost was that we wouldn't actually be using the right tracing data since the engine wasn't swapped out. Now it's a new session for the local span which is a bit weirder to work with if you share objects between scopes (not impossible though) but also has a slight benefit of allowing for more obvious limiting of transaction scopes.